src/Frontend/Fill_In_With_Postnl.php

```diff
@@ -0,0 +1,298 @@
+		$mapping = Utils::get_frontend_location_mapping();
+
+		if ( ! isset( $mapping[ $location ] ) ) {
+			return false;
+		}
+
+		[ $auto_render_key, $placement_key, $expected_placement ] = $mapping[ $location ];
+
+		return 'yes' === get_option( $auto_render_key, 'no' )
+			&& get_option( $placement_key ) === $expected_placement;
+	}
+
+	/**
+	 * Register REST API routes.
+	 *
+	 * @return void
+	 */
+	public function register_rest_routes(): void {
+		register_rest_route(
+			'postnl/v1',
+			'/get-redirect-uri',
+			array(
+				'methods'             => 'POST',
+				'callback'            => array( $this, 'handle_get_redirect_uri' ),
+				'permission_callback' => '__return_true',
+			)
+		);
+	}
+
+	/**
+	 * Handle the REST API request to get the redirect URI.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function handle_get_redirect_uri( WP_REST_Request $request ): WP_REST_Response {
+		// Check for nonce verification if needed.
+		$nonce = $request->get_header( 'X-WP-Nonce' );
+		if ( ! wp_verify_nonce( $nonce, 'wp_rest' ) ) {
+			return new WP_REST_Response(
+				array(
+					'success' => false,
+					'message' => esc_html__( 'Invalid or expired nonce.', 'postnl-for-woocommerce' ),
+				),
+				403
+			);
+		}
+
+		if ( ! $this->settings->is_fill_in_with_postnl_enabled() ) {
+			return new WP_REST_Response(
+				array(
+					'success' => false,
+					'message' => esc_html__( 'Fill in with PostNL is not enabled or Client ID is missing.', 'postnl-for-woocommerce' ),
+				),
+				400
+			);
+		}
+
+		$code_verifier = bin2hex( random_bytes( 32 ) );
+		set_transient( 'postnl_' . self::$session_verifier_key, $code_verifier, 600 ); // 10 min TTL
+
+		$state = bin2hex( random_bytes( 32 ) );
+		// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+		$code_challenge = rtrim( strtr( base64_encode( hash( 'sha256', $code_verifier, true ) ), '+/', '-_' ), '=' );
+		$redirect_uri   = $this->settings->get_redirect_uri( $code_challenge, $state );
+
+		return new WP_REST_Response(
+			array(
+				'success' => true,
+				'data'    => array(
+					'redirect_uri' => $redirect_uri,
+				),
+			),
+			200
+		);
+	}
+
+	/**
+	 * Enqueue scripts.
+	 */
+	public function enqueue_scripts(): void {
+		$postnl_checkout_params = array(
+			'rest_url'   => rest_url( 'postnl/v1/get-redirect-uri' ),
+			'ajax_url'   => admin_url( 'admin-ajax.php' ),
+			'nonce'      => wp_create_nonce( 'wp_rest' ),
+			'ajax_nonce' => wp_create_nonce( 'postnl_user_info' ),
+		);
+
+		wp_enqueue_script(
+			'fill-in-with-postnl',
+			POSTNL_WC_PLUGIN_DIR_URL . '/assets/js/fill_in_with_postnl.js',
+			array( 'jquery' ),
+			'1.0',
+			true
+		);
+
+		wp_localize_script(
+			'fill-in-with-postnl',
+			'postnlCheckoutParams',
+			$postnl_checkout_params
+		);
+	}
+
+	/**
+	 * Render the "Fill in with PostNL" button using a WooCommerce template.
+	 *
+	 * @return void
+	 */
+	public function render_button(): void {
+		if ( ! $this->is_enabled() ) {
+			return;
+		}
+
```

Comment:
```suggestion
		if ( ! $this->is_enabled() ) {
			return;
		}
wp_enqueue_script( 'fill-in-with-postnl' );
```
Perhaps we can enqueue it here?